src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -748,6 +773,16 @@ public virtual void OnNodeModified(bool forceExecute = false)
             if (handler != null) handler(this);
         }
 
+        public event Action<NodeModel> ToggleNodeFreeze;
+        public virtual void OnToggleNodeFreeze()
+        {
+            if (!RaisesModificationEvents)
+                return;
+
+            var handler = ToggleNodeFreeze;
+            if (handler != null) handler(this);
+        }
```

Comment:
This is the same event as NodeModified. Just that this event causes the nodes to be deleted from AST.   Currently workspacemodel has no propertychangedevent for nodes. I dont want to introduce one there. Instead , went with event based model here.